cpp-client/deephaven/client/include/public/deephaven/client/chunk/chunk.h

```diff
@@ -0,0 +1,161 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <string_view>
+#include <variant>
+#include "deephaven/client/utility/utility.h"
+
+namespace deephaven::client::chunk {
+/**
+ * Base type for Chunks
+ */
+class Chunk {
+protected:
+  Chunk() = default;
+  explicit Chunk(size_t size) : size_(size) {}
+  Chunk(Chunk &&other) noexcept = default;
+  Chunk &operator=(Chunk &&other) noexcept = default;
+  virtual ~Chunk() = default;
+
+public:
+  size_t size() const { return size_; }
```

Comment:
Do we need to distinguish between size and capacity?  Seems like we inherit not the same, but a similar kind of ambiguity here wrt the java code.

Reply:
You explained to me over VC that we are using Chunk as a value type slice-like that essentially represents a pointer and a length; the usage pattern should reflect that whoever intends to return one should create a new one specifying the right resulting size.